src/utils/init/config-github.js

```diff
@@ -1,188 +1,215 @@
+  return {
+    name: chalk.magenta(repoName),
+    owner: chalk.magenta(repoOwner),
+  }
+}
 
 const PAGE_SIZE = 100
 
-module.exports = async function configGithub(ctx, site, repo) {
-  const { api, globalConfig } = ctx.netlify
-  const current = globalConfig.get('userId')
+const isValidToken = (token) => {
+  return token && token.user && token.token
+}
 
-  let ghtoken = globalConfig.get(`users.${current}.auth.github`)
+const getGitHubToken = async ({ log, globalConfig }) => {
+  const userId = globalConfig.get('userId')
+  const githubToken = globalConfig.get(`users.${userId}.auth.github`)
 
-  if (!ghtoken || !ghtoken.user || !ghtoken.token) {
-    const newToken = await ghauth({
-      opts: {
-        scopes: ['admin:org', 'admin:public_key', 'repo', 'user'],
-        userAgent: UA,
-        note: `Netlify CLI ${os.userInfo().username}@${os.hostname()}`,
-      },
-      log: ctx.log,
-    })
-    globalConfig.set(`users.${current}.auth.github`, newToken)
-    ghtoken = newToken
+  if (isValidToken(githubToken)) {
+    return githubToken.token
   }
-  const octokit = new Octokit({
-    auth: `token ${ghtoken.token}`,
-  })
-
-  const key = await api.createDeployKey()
-  const parsedURL = parseGitRemote(repo.repo_path)
 
-  await octokit.repos.addDeployKey({
-    title: 'Netlify Deploy Key',
-    key: key.public_key,
-    repo: parsedURL.name,
-    owner: parsedURL.owner,
-    read_only: true,
+  const newToken = await ghauth({
+    log,
   })
+  globalConfig.set(`users.${userId}.auth.github`, newToken)
+  return newToken.token
+}
 
-  repo.deploy_key_id = key.id
-
-  // TODO: Look these up and default to the lookup order
+const getGitHubClient = ({ token }) => {
+  const octokit = new Octokit({
+    auth: `token ${token}`,
+  })
+  return octokit
+}
 
-  let defaultBuildCmd
-  let defaultBuildDir = '.'
-  // read from netlify toml
-  const { build } = ctx.netlify.config
-  if (build && build.command) {
-    defaultBuildCmd = build.command
-  }
-  if (build && build.publish) {
-    defaultBuildDir = build.publish
+const addDeployKey = async ({ log, api, octokit, repoOwner, repoName, failAndExit }) => {
+  log('Adding deploy key to repository...')
+  const key = await createDeployKey({ api, failAndExit })
+  try {
+    await octokit.repos.addDeployKey({
+      title: 'Netlify Deploy Key',
+      key: key.public_key,
+      owner: repoOwner,
+      repo: repoName,
+      read_only: true,
+    })
+    log('Deploy key added!')
+    return key
+  } catch (error) {
+    let message = formatErrorMessage({ message: 'Failed adding GitHub deploy key', error })
+    if (error.status === 404) {
+      const { name, owner } = formatRepoAndOwner({ repoName, repoOwner })
+      message = `${message}. Does the repository ${name} exist and do ${owner} has the correct permissions to set up deploy keys?`
+    }
+    failAndExit(message)
   }
-  if (build && build.functions) console.log(`Netlify functions folder is ${chalk.yellow(build.functions)}`)
```

Comment:
Instead of just printing the `functions` directory the new code will now prompt for it like other settings